src/librustc_trans/mir/constant.rs

```diff
@@ -183,27 +220,25 @@ pub fn trans_static_initializer<'a, 'tcx>(
     let param_env = ty::ParamEnv::empty(traits::Reveal::All);
     ccx.tcx().const_eval(param_env.and(cid))?;
 
-    let ptr = ccx
+    let alloc_id = ccx
         .tcx()
         .interpret_interner
         .borrow()
         .get_cached(cid)
-        .expect("global not cached")
-        .to_ptr().unwrap();
-    assert_eq!(ptr.offset, 0);
+        .expect("global not cached");
 
     let alloc = ccx
         .tcx()
         .interpret_interner
         .borrow()
-        .get_alloc(ptr.alloc_id.0)
+        .get_alloc(alloc_id)
         .expect("miri alloc not found");
     Ok(global_initializer(ccx, alloc))
 }
 
 impl<'a, 'tcx> MirContext<'a, 'tcx> {
     // Old version of trans_constant now used just for SIMD shuffle
-    pub fn remove_me__shuffle_indices(&mut self,
+    pub fn remove_me_shuffle_indices(&mut self,
```

Comment:
LOL this wasn't a typo, I meant it to be an ugly name like that.

Reply:
My bad. I got an error for non-snake-case names, and fixed it by removing the `_` instead of adding a `deny` attribute. :-P

Reply:
Ah if it triggered the lint then no worries! I didn't know it cared.

Reply:
Yep, it's picky.